plugins/powerpc/powerpc_arith.ml

```diff
@@ -0,0 +1,420 @@
+
+open Powerpc.Std
+
+(** Fixed-Point Arithmetic Instructions - Negate
+    Page 72 of IBM Power ISATM Version 3.0 B
+    example:
+    7c 22 00 d0   neg  r1, r2
+    7c 22 00 d1   neg. r1, r2 *)
+let neg cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let ra = unsigned cpu.reg ops.(1) in
+  RTL.[ rt := lnot ra + one ]
+
+(** Fixed-Point Arithmetic Instructions - Substract From
+    Page 69 of IBM Power ISATM Version 3.0 B
+    example:
+    7c 22 18 50    subf  r1, r2, r3
+    7c 22 18 51    subf. r1, r2, r3 *)
+let subf cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let ra = unsigned cpu.reg ops.(1) in
+  let rb = unsigned cpu.reg ops.(2) in
+  RTL.[rt := (lnot ra) + rb + one]
+
+(** Fixed-Point Arithmetic Instructions - Substract From Immediate Carrying
+    Page 70 of IBM Power ISATM Version 3.0 B
+    example:
+    20 22 10 92    subfic r1, r2, 4242 *)
+let subfic cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let ra = unsigned cpu.reg ops.(1) in
+  let si = unsigned imm ops.(2) in
+  RTL.[
+    rt := (lnot ra) + si + one;
+    cpu.ca   := low cpu.word_width si < low cpu.word_width ra;
+    cpu.ca32 := low word si < low word ra;
+  ]
+
+(** Fixed-Point Arithmetic Instructions - Substract From Carrying
+    Page 70 of IBM Power ISATM Version 3.0 B
+    example:
+    7c 22 18 10    subfc  r1, r2, r3
+    7c 22 18 11    subfc. r1, r2, r3 *)
+let subfc cpu ops =
+  let rt = signed cpu.reg ops.(0) in
```

Comment:
now it really suspicious, why are you using unsinged operands for `subfic` and signed for `subfc` if they only differ in the sort of the third operand, i.e., register vs. immediate.